text/0005-application-formalization.md

```diff
@@ -0,0 +1,191 @@
+| id       | String  | Identifier for the module conforming to pattern: `{name}-{version}`  | e.g. "mod-orders-storage-13.5.0" |
+| name     | String  | Name of the module                                                   | e.g. "mod-orders"                |
+| version  | String  | Version of the module                                                | e.g. "13.5.0"                    |
+| url      | String  | URL pointing to the module descriptor for this module                | e.g. "https://folio-registry.dev.folio.org/_/proxy/modules/mod-orders-storage-13.5.0" |
+
+Example:
+```
+{
+  "id": "app-gobi-0.0.1",
+  "name": "app-gobi",
+  "version": "0.0.1",
+  "description": "Application facilitating the placement of orders via GOBI and other compatible vendors",
+  "dependencies": [
+    {
+      "id": "app-orders-0.0.1",
+      "name": "app-orders",
+      "version": "^0.0.1"
+    },
+    {
+      "id": "app-organizations-0.0.1",
+      "name": "app-organizations",
+      "version": "^0.0.1"
+    }
+  ],
+  "modules": [
+    {
+      "id": "mod-gobi-2.6.0",
+      "name": "mod-gobi",
+      "version": "2.6.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/mod-gobi-2.6.0"
+    }
+  ],
+  "uiModules": [
+    {
+      "id": "folio_gobi-settings-2.0.0",
+      "name": "folio_gobi-settings",
+      "version": "2.0.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/folio_gobi-settings-2.0.0"
+    }
+  ]
+}
+```
+**N.B.** Full module descriptors and metadata have been omitted for readability reasons.
+
+## Benefits
+* Allows for expanding the bounded context
+  * By expanding the bounded context from the module to the Application, multiple modules can directly access the storage of the microservice.
+  * Benefites of this include:  
+    * Improved performance and simplified logic.  Modules can now perform the same operations with fewer API calls.
+    * Fewer modules since this eliminates the need for separate business logic and storage modules.
+* This is the first step toward breaking the monolithic release into application-specific releases
+* Results in fewer things to work with at the system operator level
+* Brings us closer to realizing the idea of having an application store/marketplace
+
+## Risks and Drawbacks
+* Folio has too many module dependencies which must be reduced
+* Transition to Applications will need to be iterative
+  * First pass is to create larger applications spanning multiple areas in order to satisfy dependencies
+  * Subsequent passes will further break up these large applications into smaller, applications focused on a single area of functionality
+* Since Applications are a vertical slice of functionality, better alignment/coordination between back-end and front-end development teams is required.
+  * Reorganizing development teams around applications would help
```

Comment:
I agree reorganizing development teams around applications would help address this risk.  However it creates other risks.   From [TC questions](https://docs.google.com/document/d/1OLae-eGN3_dEOXbAr7Tr1cvudBzqNA7TttzH9QmvuR8/edit#heading=h.50f0f2jmdfjl):

> If we “Line-up Applications dev team ownership/responsibility” (p.25), what implications are there on the ability to contribute development by smaller development teams or teams which do not “own” modules

Reply:
Is this applicable to the scope/general direction of the RFC?  Is this request better suited to a later stage?

Reply:
Definitely not scope/direction-related, but (like above) reacting to the risk.  Fine to defer this, or put it in unanswered questions, as you like.

Reply:
Let's resolve this in scope of the PRELIMINARY REVIEW stage since it's not about scope/general direction.  I will address this in the DRAFT REFINEMENT stage.